### PR TITLE
Stop passing around a single logging instance

### PR DESCRIPTION
**Pull request by jeremycline**
This configures logging and stops passing around a single logger
instance.

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

### CODE REVIEW

File: anitya/app.py [hunk 1]
```diff
@@ -38,22 +37,46 @@
 APP.oid = OpenID(APP)
 
 # Set up the logging
-if not APP.debug:
-    APP.logger.addHandler(anitya.mail_logging.get_mail_handler(
-        smtp_server=APP.config.get('SMTP_SERVER', '127.0.0.1'),
-        mail_admin=APP.config.get('MAIL_ADMIN', 'admin@fedoraproject.org')
+logging_config = {
```
Comment (pypingou) [inline anitya/app.py]: What about placing this in the configuration file?

**Comment by pypingou** [comment]
👍 with one question

File: anitya/app.py (hunk 1 above)
Comment (jeremycline) [inline anitya/app.py]: While that does allow for maximum configuration, when I wrote this I thought that maybe it's more user-friendly to have one log level setting. I don't mind either way so if you think it's better to expose all this in the config I can switch it.

Comment (pypingou) [inline anitya/app.py]: I don't mind either way, it's just that I've often seen this dict provided as configuration.

Anyway, we can always start like this and reconsider later if needed.

**Comment by pypingou** [comment]
Still 👍

**Pull request merged**